addons/uh60_weapons/functions/fnc_fired.sqf

```diff
@@ -16,4 +16,84 @@ if (_ammo == "VTX_Hellfire_AGM114K" || _ammo == "VTX_Hellfire_AGM114N") then {
 
 if (player == _gunner && local _gunner) then {
 	[_vehicle] call vtx_uh60_weapons_fnc_updateMFDValues;
-};
\ No newline at end of file
+};
+
+//spaggetti code:  Mix of Yax's ITC Air APKs and HARM Guidence by Donov
+if (_ammo == "ammo_Missile_HARM_HL" && vehicle _gunner == _vehicle) then {
+	_index = 0;	//reset index to enter while loop
+	while {_index < 4} do {
+		if ((getPylonMagazines _vehicle # _index) == "PylonMissile_Missile_HARM_x1_HL") then {
+			if (_vehicle ammoOnPylon (_index + 1) == 0) then { _vehicle setPylonLoadout [(_index + 1), "", true]; /*systemChat "Reset Pylon"*/};
+			//This loop checks for which HARM was fired, and replaces the pylon with nothing.  It removes the missile
+		};
+		_index = _index + 1;  //python be like
+	};
+	[{	//start of Yax's ITC APK Code
+		(_this select 0) params ["_projectile", "_ammo", "_foundTarget", "_launchTime","_lastFrameTime", "_targetPos"];
+		//Target pos needs to get passed as a parameter or this does not work
+		private _frameTime = time - _lastFrameTime;
+		(_this select 0) set [4, time];
+
+		if (!alive _projectile) exitWith {
+			[_this select 1] call CBA_fnc_removePerFrameHandler;
+		};
+
+		//Start of Donov HARM Guidence computer
+		systemChat "Missile Alive";
+		_missileTgtPos = [getPosASL _projectile, vectorDir _projectile, _projectile, vehicle _gunner] call ace_laser_fnc_shootRay;
+		/*  Using Shoot ray to determine a spot in space where the missile is looking in front of it 
+		 *  Then using nearestObjects to pull a list of "AllVehicles" from the radius of the seekerhead
+		 *  Then sorting through the list of "AllVehicls" and setting the target as the first vehicle (closest) in the list
+		 *		Only choosing vehicles that have radar on (turning off radar and moving should save the vehicle (in theory))
+		*/
+		if (!_foundTarget) then {
+			if ((_missileTgtPos # 1) != 0) then {
+				_seenObjects = nearestObjects [(_missileTgtPos # 0), ["AllVehicles"], (_missileTgtPos # 1)*5.6712818 + 10];
```

Comment:
nearestObjects is probably too performance intensive, do a performance comparison using the `vehicles` array, with 100 vehicles present

Reply:
`_sortList = vehicles; 
_checkPos = (getPosASL player);
_rerun = 1;

while {_rerun > 0} do {
_rerun = 0;
for "_i" from 0 to ((count _sortList) - 1) do {
  _tempPosA = (getPosASL (_sortList # _i));
  _tempPosB = (getPosASL (_sortList # (_i+1)));
  _distA = _tempPosA distance _checkPos;
  _distB = _tempPosB distance _checkPos;

  if (_distA > _distB) then {
    _sortList append [(_sortList # _i)];
    _sortList deleteAt _i;
    _rerun = _rerun + 1;
  };
};
};

_sortList`


Sorting Algorythim I was going to use creates more lag (0.7333676ms) than nearestObjects (0.0071ms)